spatialmath/box.go

```diff
@@ -352,3 +352,102 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// ToPointCloud converts a box geometry into a []r3.Vector.
+func (b *box) ToPointCloud(options map[string]interface{}) []r3.Vector {
+	// check for user defined spacing
+	var faces []r3.Vector
+	var iter float64
+	if options["resolution"] != nil {
+		iter = options["resolution"].(float64)
+	} else {
+		iter = 0.15 // default spacing
```

Comment:
This should be a `const` at the top of a file in this package

Reply:
gotcha

Reply:
actually, I do not think it would make sense to have `iter` be a constant value. This is because the default value for sphere and box point density are not the same.

Reply:
The key piece here is "magic numbers shouldn't exist in the middle of code somewhere, they should be defined as consts at the top of the file". Totally fine if we want to have `defaultBoxPointDensity` as well as `defaultSpherePointDensity`.

However, both point densities should approximate to "points per square mm of geometry surface area" or something of that sort, right? Why are we using different densities?

Reply:
I do not think _both_ point densities approximate "points per square mm of geometry surface area". For a sphere the `iter` value tells us how many points **total** should lie on the surface, whereas _only_ for a box the `iter` value translates to points per square mm of geometry surface area